projects/max14906/src/examples/basic/basic_example.c

```diff
@@ -0,0 +1,194 @@
+	if (ret)
+		goto error;
+
+	/** Global Interrupt Controller */
+	struct no_os_irq_ctrl_desc *global_desc;
+	struct no_os_irq_init_param global_desc_param = {
+		.irq_ctrl_id = GPIO_IRQ_ID,
+		.platform_ops = &max_irq_ops,
+		.extra = NULL
+	};
+
+	ret = no_os_irq_ctrl_init(&global_desc, &global_desc_param);
+	if (ret)
+		goto global_irq_error;
+
+	/** GPIO Interrupt Controller */
+	struct no_os_irq_ctrl_desc *gpio_irq_desc;
+	struct no_os_irq_init_param gpio_irq_desc_param = {
+		.irq_ctrl_id = GPIO_IRQ_ID,
+		.platform_ops = GPIO_IRQ_OPS,
+		.extra = NULL
+	};
+
+	ret = no_os_irq_ctrl_init(&gpio_irq_desc, &gpio_irq_desc_param);
+	if (ret)
+		goto gpio_irq_error;
+
+	struct no_os_callback_desc gpio_cb = {
+		.callback = gpio_callback_fn,
+		/** Parameter to be passed when the callback is called. */
+		.ctx = max14906_desc,
+		.event = NO_OS_EVT_GPIO,
+		.peripheral = NO_OS_GPIO_IRQ,
+		.handle = NULL
+	};
+
+	ret = no_os_irq_register_callback(gpio_irq_desc, GPIO_FAULT_PIN_NUM,
+					  &gpio_cb);
+	if (ret)
+		goto gpio_irq_error;
+
+	ret = no_os_irq_trigger_level_set(gpio_irq_desc, GPIO_FAULT_PIN_NUM,
+					  NO_OS_IRQ_EDGE_BOTH);
+	if (ret)
+		goto gpio_irq_error;
+
+	ret = no_os_irq_set_priority(gpio_irq_desc, GPIO_FAULT_PIN_NUM, 1);
+	if (ret)
+		goto gpio_irq_error;
+
+	ret = no_os_irq_enable(gpio_irq_desc, GPIO_FAULT_PIN_NUM);
+	if (ret)
+		goto gpio_irq_error;
+
+	ret = no_os_irq_enable(global_desc, GPIO0_IRQn);
+	if (ret)
+		goto gpio_irq_error;
+
+	/** Setting SLED set bit 1 in the config register. */
+	ret = max149x6_reg_update(max14906_desc, MAX14906_CONFIG1_REG,
+				  MAX14906_SLED_MASK, no_os_field_prep(MAX14906_SLED_MASK, 1));
+	if (ret)
+		goto gpio_irq_error;
+
+	/** Turning the Status LEDs on, then off. */
+	for(i = 0; i <  MAX14906_CHANNELS; i++) {
+		ret = max149x6_reg_update(max14906_desc, MAX14906_SETLED_REG,
+					  MAX14906_SLED_CH(i), no_os_field_prep(MAX14906_SLED_CH(i), 1));
+		if (ret)
+			goto gpio_irq_error;
+
+		no_os_mdelay(500);
+
+		ret = max149x6_reg_update(max14906_desc, MAX14906_SETLED_REG,
+					  MAX14906_SLED_CH(i), no_os_field_prep(MAX14906_SLED_CH(i), 0));
+		if (ret)
+			goto gpio_irq_error;
+
+		no_os_mdelay(500);
+	}
+
+	/** Setting a current limit for channel 0. */
+	ret = max14906_climit_set(max14906_desc, 0, climit);
+	if(ret)
+		goto gpio_irq_error;
+
+	/** Read current limit for all channels. */
+	for(i = 0; i < MAX14906_CHANNELS; i++) {
+		ret = max14906_climit_get(max14906_desc, i, &climit2);
+		if(ret)
+			goto gpio_irq_error;
+		pr_info("Current limit for channel %d is %s\n", i,
+			climit_name[climit2]);
+	}
+
+	/** Setting the on state for channel 0, and then verifying it. */
+	if(max14906_desc->en_gpio) {
+		ret = max14906_ch_set(max14906_desc, 0, 1);
+		if(ret)
+			goto gpio_irq_error;
+
+		ret = max14906_ch_get(max14906_desc, 0, &val);
+		if(ret)
+			goto gpio_irq_error;
+	}
+
+	return 0;
+
+gpio_irq_error:
+	gpio_desc->platform_ops->gpio_ops_remove;
```

Comment:
why not use no_os_irq_ctrl_remove instead?... please make sure the _remove APIs are called only if init was successful and an error occurred after, see all cases where it is applicable